src/sagemaker/model.py

```diff
@@ -912,6 +906,14 @@ def _create_sagemaker_model(
             )
             self.sagemaker_session.create_model(**create_model_args)
 
+    def _get_model_uri(self):
```

Comment:
can you clarify the change? As far as I understand we moved it to a function. Not sure how it fixes https://github.com/aws/sagemaker-python-sdk/issues/4248

Reply:
I refactored the model_uri computation into a function because it's now used in two places: when creating the model and when deploying it.

It used to not be computed correctly when deploying the model. Making use of this function when deploying fixes the issue.

Reply:
Thanks. Can you add an unit test?

Reply:
Done!